RegressionModels/LinearRegression/HousePriceEstimation.py

```diff
@@ -2,173 +2,82 @@
-
-#to find columns whcih caontaing more than 70% of single values
-for i in Data.columns:
-    print(Data[i].value_counts()[(((Data[i].value_counts())/Data.shape[0]*100)>70)])
-
-#this coloumn contains more than 70% of the single values
-Data_updated = Data_updated.drop(['MSZoning','LandContour','Condition1','BldgType','RoofStyle','RoofMatl','ExterCond','LotConfig','LandSlope','Condition2','BsmtCond','BsmtFinType2','Heating','Electrical','LowQualFinSF','BsmtHalfBath','KitchenAbvGr','Functional','GarageQual','GarageCond','PavedDrive','EnclosedPorch','3SsnPorch','PoolArea','MiscVal','SaleType','SaleCondition'],axis=1)
+null_value_sum = Raw_data.isnull().sum()
+total_num_rows = Raw_data.shape[0]
 
 #missing values more than 50 percentage
-num_missing = (Data_updated[Data_updated.columns] == 0).sum()
-#checking coloummns which conatin more than 50% of values
-missing_value_columns =  Data_updated.columns[num_missing>500]
-Data_updated = Data_updated.drop(missing_value_columns,axis=1)
-
-#removing coloumns which contain more null values 
-Data_updated = Data_updated.drop('FireplaceQu',axis=1)
-
-
-#filling nan values with there coloumn mean
-Data_updated.fillna(Data_updated.mean(), inplace=True)
-print(Data_updated.shape)
-
-#labeling categorial variabes with numbers
-labels = Data_updated['LotShape'].astype('category').cat.categories.tolist()
-replace_map_comp = {'LotShape' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-#making a copy of the data frame
-Data_updated_replace = Data_updated.copy()
-
-#replacing the categorial varialbs
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['Neighborhood'].astype('category').cat.categories.tolist()
-replace_map_comp = {'Neighborhood' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['HouseStyle'].astype('category').cat.categories.tolist()
-replace_map_comp = {'HouseStyle' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['Exterior1st'].astype('category').cat.categories.tolist()
-replace_map_comp = {'Exterior1st' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['Exterior2nd'].astype('category').cat.categories.tolist()
-replace_map_comp = {'Exterior2nd' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['MasVnrType'].astype('category').cat.categories.tolist()
-replace_map_comp = {'MasVnrType' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
+num_missing = (Raw_data[Raw_data.columns] == 0).sum()
```

Comment:
check this one it seems wrong

Raw_data.columns[(Raw_data.isnull().sum()/Raw_data.shape[0])>0.7]